File: assignments/a1/q9/Java/Tokener.java
//Tokener:top//
//Tokener:import//
import java.util.*;

// <rule>:Tokener ::= <TOKEN> <ID> <RE>
public class Tokener extends Rule /*Tokener:class*/ {

    public static final String $className = "Tokener";
    public static final String $ruleString =
        "<rule>:Tokener ::= <TOKEN> <ID> <RE>";

    public Token token;
    public Token id;
    public Token re;

    public Tokener(Token token, Token id, Token re) {
//Tokener:init//
        this.token = token;
        this.id = id;
        this.re = re;
    }

    public static Tokener parse(Scan scn$, Trace trace$) {
        if (trace$ != null)
            trace$ = trace$.nonterm("<rule>:Tokener", scn$.lno);
        Token token = scn$.match(Token.Match.TOKEN, trace$);
        Token id = scn$.match(Token.Match.ID, trace$);
        Token re = scn$.match(Token.Match.RE, trace$);
        return new Tokener(token, id, re);
    }

//Tokener//
}

File: assignments/a1/q9/Python/Tokener.py
#Tokener:top#
#Tokener:import#

# <rule>:Tokener ::= <TOKEN> <ID> <RE>
class Tokener(Rule): #Tokener:class#

    className = "Tokener"
    ruleString = "<rule>:Tokener ::= <TOKEN> <ID> <RE>"
    token = None
    id = None
    re = None

    def __init__(token, id, re):
        #Tokener:init#
        self.token = token
        self.id = id
        self.re = re

#Tokener#
